cmd/clusterctl/client/yamlprocessor/template_processor_simple_test.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package yamlprocessor
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/test"
+)
+
+func TestSimpleProcessor(t *testing.T) {
+
+	t.Run("returns expected artifact name", func(t *testing.T) {
+		g := NewWithT(t)
+		p := NewSimpleProcessor()
+		g.Expect(p.ArtifactName("some-version", "some-flavor")).To(Equal("cluster-template-some-flavor.yaml"))
+		g.Expect(p.ArtifactName("", "")).To(Equal("cluster-template.yaml"))
+	})
+
+	t.Run("returns the variables from the yaml", func(t *testing.T) {
```

Comment:
Yup...makes sense. Since there is no shared setup/teardown code as per the go testing patterns it makes sense to split them into separate Test funcs. For others context: https://blog.golang.org/subtests

I'll update this.